sdk/core/core-lro/src/poller.ts

```diff
@@ -0,0 +1,32 @@
+import { PollOperation } from "./pollOperation";
+
+export abstract class Poller<T> {
+  public stopped?: boolean;
+  public operation: PollOperation<T>; 
+  public promise: Promise<void>;
+
+  constructor(
+    operation: PollOperation<T>,
+    stopped?: boolean,
+  ) {
+    this.operation = operation;
+    this.stopped = stopped;
+    this.promise = this.poll();
```

Comment:
I kinda still like the idea of a `done` promise specifically, which is separate than the promise returned by each invocation of `poll`.

It also doesn't seem (from the examples given) that `poll` is designed to support manually polling in this current format.

Reply:
this promise is the done promise! I just added a pollOnce function, which should clarify this.

Reply:
Hmm... so stopping also completes it.